### ISSUE TITLE
Arbitrage - Buying high and selling low

### ISSUE THREAD

**Issue by john-hummingbot-support** [CoinAlpha/hummingbot#3106]
**Describe the bug**
// A clear and concise description of what the bug is.
Discord user einstein9936 is running an arbitrage bot. the bot was able to spot an opportunity to buy in binance at price $55967.52 and sell on kucoin at price $56362.1

![image](https://user-images.githubusercontent.com/78332435/112203534-9d0a4b00-8c4d-11eb-930f-261e77ab6e99.png)

however the bot executed the opposite

the bot buys in binance at price $56362.1
![image](https://user-images.githubusercontent.com/78332435/112203718-c925cc00-8c4d-11eb-9775-e8853458561a.png)

and executes a sell order in kucoin at price $55967.52

![image](https://user-images.githubusercontent.com/78332435/112203796-de9af600-8c4d-11eb-8e08-2ad50e34d79b.png)

according to the user, the display on the open opportunity was actually reversed which is why the bot executes the trades in reverse as well. 

Talked with Daniel and according to him this should be a P1 bug, the user failed to create a github ticket so I'm creating this on his behalf. The whole log provided by the user is attached here and the whole conversation can be found here https://discord.com/channels/530578568154054663/563462222878212098/822173151856951336 .

[message (3).txt](https://github.com/CoinAlpha/hummingbot/files/6192160/message.3.txt)

**Comment by dennisocana** [comment]
@john-hummingbot-support can you include the configs in the report

**Comment by john-hummingbot-support** [comment]
Hi @dennisocana, the user agreed that he will file a github ticket with regards to this so i instructed him to attach the whole log, config and the screenshot which is why i was not able to get his config. I already sent him a follow-up message. I will post it here once i get the config. thank you!

**Comment by john-hummingbot-support** [comment]
Here are all the related files and logs provided by the user. thank you!

![image](https://user-images.githubusercontent.com/78332435/112486995-5c2e4580-8db7-11eb-8566-f48410cce870.png)


[Archive.zip](https://github.com/CoinAlpha/hummingbot/files/6205242/Archive.zip)

**Comment by shankinson** [comment]
I also ran into this same issue today when running the arbitrage bot.

On one exchange I was using the ETH-USDT pair and on the other, I was using the ETH-CAD pair. 

For exchange one, the best bid was 2056.72 and the best ask was 2056.73. For exchange two the best bid was 2611.56 and the best ask 2612.39.

The secondary_to_primary_base_conversion_rate was set to 1.0 and secondary_to_primary_quote_conversion_rate was set to 0.795228628230616.

If you take the ask on exchange one ( 2056.73 / 0.795228628230616 = 2586.34 ) and take the bid (2611.56) on exchange 2 there is about a 1% arbitrage (2611.56 - 2586.34 = 25.22). 

Humming Bot tried to execute the order, but it seemed to mix up the prices:
```
Error submitting BUY LIMIT order to Exchange 1 for 0.10870000 ETH-USDT 2611.56000000
``` 
^ didn't have enough USDT
```
EVENT_LOG - {"timestamp": 1617535675.0, "type": "OrderType.LIMIT", "trading_pair": "ETH-CAD", "amount": "0.1087", "price": "2056.73", "order_id": "7131617535675009", "exchange_order_id": "14984491360", "leverage": 1, "position": "NILL", "event_name": "SellOrderCreatedEvent", "event_source": "Exchange 2"}
```

It tried to buy on exchange 1 with the price from exchange 2 and sell on exchange 2 with the price from exchange 1.

I looked through the code and I think the buy_price and sell_price are reversed here; https://github.com/CoinAlpha/hummingbot/blob/master/hummingbot/strategy/arbitrage/arbitrage.pyx#L349

The buy price is being set to the bid_price and the sell price is being set to the ask_price (https://github.com/CoinAlpha/hummingbot/blob/master/hummingbot/strategy/arbitrage/arbitrage.pyx#L524), however, since we are being takers in this scenario we should be selling at the bid and buying at the ask, wouldn't we?

**Comment by shankinson** [comment]
I just opened a [https://github.com/CoinAlpha/hummingbot/pull/3158](PR) with the change

**Comment by shankinson** [comment]
Just noting that with the change I was able to successfully issue an arbitrage trade last night.

Exchange 1 had a best bid of 2031.37 and best ask of 2031.38. Exchange 2 had a best bid of 2577.54 25 and a best ask of 2580.18.

HummingBot log shows `Executing limit order buy of ETH-USDT at Exchange 1 for 2031.38000000 and sell of ETH-CAD at Exchange 2 for 2575.00 with amount 0.03720000, and profitability 1.006010958398749042464156958 [clock=2021-04-05 03:50:45+00:00]`

2575 - 2031.38 /  0.795228628230616 = 20.54.

I can see that both orders filled correctly on their exchanges, making the arbitrage profit.

**Comment by shankinson** [comment]
Yes, I've noticed the same issue even during live trading and with different algorithms. My account value goes up but it does not map to the values shown in the status bar, so there is some kind of error in those calculations.

There are several issues opened around the history command also reporting bad values (which I have also noticed). I've resorted to tracking initial starting values with paper/pen and spot-checking things throughout the day while the bot is running until I get time to dig into the code.

Issues: #3094, #3093, #3021
